src/Repository/EmailSubscriptionHistoryRepository.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace AppBundle\Repository;
+
+use AppBundle\Entity\Adherent;
+use AppBundle\Entity\Reporting\EmailSubscriptionHistory;
+use AppBundle\Entity\Reporting\EmailSubscriptionHistoryAction;
+use Cake\Chronos\Chronos;
+use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
+use Doctrine\Common\Persistence\ManagerRegistry;
+
+class EmailSubscriptionHistoryRepository extends ServiceEntityRepository
+{
+    public function __construct(ManagerRegistry $registry)
+    {
+        parent::__construct($registry, EmailSubscriptionHistory::class);
+    }
+
+    /**
+     * Count active adherents (not users) history for each specified subscription types in the referent managed area before the specified date.
+     */
+    public function countAllByTypeForReferentManagedArea(Adherent $referent, array $subscriptionsTypes, \DateTimeInterface $until): array
+    {
+        if (!$referent->isReferent()) {
+            throw new \InvalidArgumentException('Adherent must be a referent.');
+        }
+
+        $qb = $this->createQueryBuilder('history')
+            ->select('history.adherentUuid, history.subscribedEmailType, history.action, COUNT(history) AS count')
+            ->where('history.referentTag IN (:tags)')
+            ->andWhere('history.subscribedEmailType IN (:subscriptions)')
+            ->andWhere('history.date <= :until')
```

Comment:
do you have an index for this column?

Reply:
Yes -> https://github.com/EnMarche/en-marche.fr/pull/2383/files#diff-82b04cbbba96b1c98b716e7a91d4f82eR19